src/main/java/com/google/maps/GeoApiContext.java

```diff
@@ -62,49 +83,40 @@
-                                                          Integer maxRetries,
-                                                          ExceptionsAllowedToRetry exceptionsAllowedToRetry);
-
-    <T, R extends ApiResponse<T>> PendingResult<T> handlePost(String hostName, String url, String payload,
-                                                              String userAgent, Class<R> clazz,
-                                                              FieldNamingPolicy fieldNamingPolicy, long errorTimeout,
-                                                              Integer maxRetries,
-                                                              ExceptionsAllowedToRetry exceptionsAllowedToRetry);
-
-    void setConnectTimeout(long timeout, TimeUnit unit);
-    void setReadTimeout(long timeout, TimeUnit unit);
-    void setWriteTimeout(long timeout, TimeUnit unit);
-    void setQueriesPerSecond(int maxQps);
-    @Deprecated void setQueriesPerSecond(int maxQps, int minimumInterval);
-    void setProxy(Proxy proxy);
-  }
 
-  private long errorTimeout = DEFAULT_BACKOFF_TIMEOUT_MILLIS;
+    <T, R extends ApiResponse<T>> PendingResult<T> handle(String hostName, String url,
+        String userAgent, Class<R> clazz,
+        FieldNamingPolicy fieldNamingPolicy, long errorTimeout,
+        Integer maxRetries,
+        ExceptionsAllowedToRetry exceptionsAllowedToRetry);
 
-  /**
-   * Construct a GeoApiContext with OkHttp.
-   */
-  public GeoApiContext() {
-    this(new OkHttpRequestHandler());
-  }
+    <T, R extends ApiResponse<T>> PendingResult<T> handlePost(String hostName, String url,
+        String payload,
+        String userAgent, Class<R> clazz,
+        FieldNamingPolicy fieldNamingPolicy, long errorTimeout,
+        Integer maxRetries,
+        ExceptionsAllowedToRetry exceptionsAllowedToRetry);
+
+    interface Builder {
+
+      void connectTimeout(long timeout, TimeUnit unit);
+
+      void readTimeout(long timeout, TimeUnit unit);
+
+      void writeTimeout(long timeout, TimeUnit unit);
+
+      void queriesPerSecond(int maxQps);
+
+      void proxy(Proxy proxy);
+
+      RequestHandler build();
+    }
 
-  /**
-   * Construct a GeoApiContext with the specified strategy for handling requests.
-   *
-   * @see OkHttpRequestHandler
-   * @see GaeRequestHandler
-   *
-   * @param requestHandler How to handle URL requests to the Google Maps APIs.
-   */
-  public GeoApiContext(RequestHandler requestHandler) {
-    this.requestHandler = requestHandler;
-    this.exceptionsAllowedToRetry.add(OverQueryLimitException.class);
   }
 
+
   <T, R extends ApiResponse<T>> PendingResult<T> get(ApiConfig config, Class<? extends R> clazz,
-                                                     Map<String, String> params) {
+      Map<String, String> params) {
     if (channel != null && !channel.isEmpty() && !params.containsKey("channel")) {
       params.put("channel", channel);
```

Comment:
This isn't my code, but I did sign off on bringing it into the code base. At this point I think I'll stick a TODO on it and think about how best to clean it up.